testsuite/tests/apicast/policy/test_token_introspection_threescale_endpoint.py

```diff
@@ -0,0 +1,76 @@
+"""
+Test for https://issues.redhat.com/browse/THREESCALE-10591
+"""
+
+import pytest
+
+from testsuite import rawobj
+from testsuite.rhsso.rhsso import OIDCClientAuthHook
+
+pytestmark = [pytest.mark.nopersistence]
+
+
+@pytest.fixture(scope="module", autouse=True)
+def rhsso_setup(lifecycle_hooks, rhsso_service_info):
+    """Have application/service with RHSSO auth configured"""
+    lifecycle_hooks.append(OIDCClientAuthHook(rhsso_service_info))
+
+    return rhsso_service_info
+
+
+@pytest.fixture(scope="module")
+def access_token(application, rhsso_service_info):
+    """get RHSSO access token"""
+    return rhsso_service_info.access_token(application)
+
+
+@pytest.fixture(scope="module")
+def client(api_client):
+    """
+    Returns api_client without auth session set to None
+    Auth session needs to be None when we are testing access_token
+    """
+    api_client = api_client()
+
+    api_client.auth = None
+    return api_client
+
+
+@pytest.fixture
+def update_policies(service):
+    """
+    Adds the token_introspection policy configured as follows
+    """
+
+    policy_setting = rawobj.PolicyConfig(
+        "token_introspection",
+        {
+            "auth_type": "use_3scale_oidc_issuer_endpoint",
+            "max_ttl_tokens": 10,
+            "max_cached_tokens": 10,
+        },
+    )
+
+    service.proxy.list().policies.append(policy_setting)
+    service.proxy.deploy()
+
+
+# pylint: disable=unused-argument
+@pytest.mark.nopersistence  # Test checks changes during test run hence is incompatible with persistence plugin
+@pytest.mark.issue("https://issues.redhat.com/browse/THREESCALE-10591")
+def test_rhsso_logout(client, access_token, rhsso_service_info, update_policies):
+    """
+    Makes a request using rhsso auth.
+    Asserts that the request passed.
+    Logs out the user from rhsso.
+    Makes another request using rhsso auth.
+    Asserts that the request fails
+    """
+
+    response = client.get("/get", headers={"authorization": "Bearer " + access_token})
```

Comment:
convention is to use the header name in the form  `Authorization` , please use f string for string interpolation